File: backend/bookingmovieticketapp/src/main/java/com/project/bookingmovieticketapp/DTOs/MovieDTO.java
package com.project.bookingmovieticketapp.DTOs;

import lombok.*;

import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class MovieDTO {
    private String name;
    private String description;
    private int duration;
    private Date releasedate;
    private String posterurl;
    private String bannerurl;
    private String agerating;
    private float voteaverage;
    private String director;
}

File: webadmin/src/app/dtos/movie.dto.ts
export class Movie{
    name: string;
    posterurl: string;
    releasedate: string;
  voteaverage: number;

  constructor(data: any) {
    this.name = data.name || 'Unknown Title';
    this.posterurl = data.posterurl || 'no_image';
    this.releasedate = data.releasedate || 'Unknown Release Date';
    this.voteaverage = data.voteaverage || 0;
  }
}